torch_xla/csrc/BUILD

```diff
@@ -0,0 +1,289 @@
+    hdrs = [
+        ":LazyIr.h",
+        "tensor_methods.h",
+        "tensor_ops.h",
+        ":LazyNonNativeIr.h",
+        ":XLANativeFunctions.h",
+        "xla_op_builder.h",
+        "resize_ops.h",
+        "nms_op.h",
+        "aten_xla_bridge.h",
+        "convert_ops.h",
+        "convolution.h",
+        "data_ops.h",
+        "cross_replica_reduces.h",
+        "softmax_builder.h",
+        "debug_util.h",
+        "elementwise.h",
+        "generated_file_include.h",
+        "helpers.h",
+        "aten_autograd_ops.h",
+        "xla_graph_executor.h",
+        "ir.h",
+        "ir_dump_util.h",
+        "ir_util.h",
+        "lowering_context.h",
+        "matrix.h",
+        "nll_loss.h",
+        "op_by_op_executor.h",
+        "batch_norm.h",
+        "pooling.h",
+        "random.h",
+        "reduction.h",
+        "tensor.h",
+        "tensor_impl.h",
+        "tensor_util.h",
+        "token_handler.h",
+        "torch_util.h",
+        "view.h",
+        "xla_backend_impl.h",
+        "xla_lower_util.h",
+        "xla_sharding_util.h",
+    ] + glob(["ops/*.h"]),
+    deps = [
+        ":aten_cpu_fallback",
+        ":computation",
+        ":device",
+        ":einsum_utilities",
+        ":ir_builder",
+        ":layout_manager",
+        ":shape_builder",
+        "//third_party/xla_client:async_task",
+        "//third_party/xla_client:computation_client",
+        "@com_google_absl//absl/hash",
+        "@com_google_absl//absl/memory",
+        "@com_google_absl//absl/strings",
+        "@com_google_absl//absl/types:optional",
+        "@com_google_absl//absl/types:span",
+        "@org_tensorflow//tensorflow/compiler/tf2xla/kernels:conv_op_helpers",
+        "@org_tensorflow//tensorflow/compiler/xla:comparison_util",
+        "@org_tensorflow//tensorflow/compiler/xla:literal_util",
+        "@org_tensorflow//tensorflow/compiler/xla:permutation_util",
+        "@org_tensorflow//tensorflow/compiler/xla:shape_util",
+        "@org_tensorflow//tensorflow/compiler/xla:types",
+        "@org_tensorflow//tensorflow/compiler/xla/client:xla_builder",
+        "@org_tensorflow//tensorflow/compiler/xla/client/lib:arithmetic",
+        "@org_tensorflow//tensorflow/compiler/xla/client/lib:comparators",
+        "@org_tensorflow//tensorflow/compiler/xla/client/lib:constants",
+        "@org_tensorflow//tensorflow/compiler/xla/client/lib:logdet",
+        "@org_tensorflow//tensorflow/compiler/xla/client/lib:math",
+        "@org_tensorflow//tensorflow/compiler/xla/client/lib:slicing",
+        "@org_tensorflow//tensorflow/compiler/xla/stream_executor:dnn",
+        "@org_tensorflow//tensorflow/core:lib",
+        "@org_tensorflow//tensorflow/core/framework:tensor_shape",
+        "@org_tensorflow//tensorflow/core/kernels:conv_ops",
+        "@org_tensorflow//tensorflow/core/lib/core:errors",
+        "@org_tensorflow//tensorflow/core/lib/gtl:inlined_vector",
+        "@org_tensorflow//tensorflow/core/profiler/lib:traceme",
```

Comment:
same question, are these `tensorflow/core`'s dep are needed? are they used directly or generally?

we would migrate to pull XLA from OpenXLA, so would delete TensorFlow, and will delete `tensorflow/core` deps too

Reply:
Yeah, once we remove the logical dependency to tensorflow, then these will not be needed any more.